src/core/thread/mle.cpp

```diff
@@ -5515,6 +5518,58 @@ Error Mle::RxMessage::ReadRouteTlv(RouteTlv &aRouteTlv) const
 }
 #endif
 
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+void Mle::AddWakeupParent(const Mac::ExtAddress &aParent, TimeMilli aAttachTime, uint32_t aAttachWindowMs)
+{
+    CslNeighbor *parent = nullptr;
+
+    parent = mCslNeighborTable.GetNewCslNeighbor();
+    parent->SetExtAddress(aParent);
+
+    mWakeupParentAttachTime   = aAttachTime;
+    mWakeupParentAttachWindow = aAttachWindowMs;
+}
+
+CslNeighbor *Mle::GetWakeupParent(void) { return mCslNeighborTable.GetFirstCslNeighbor(); }
+
+void Mle::AttachToWakeupParent()
+{
+    CslNeighbor *parent = GetWakeupParent();
+
+    OT_ASSERT(parent != nullptr);
+
+    // TODO: Implement the logic to attach to the wakeup parent.
```

Comment:
I guess there is more code to follow since I don't see the ParentRequest actually being sent. Right? Or am I missing something?

Reply:
There is. In the initial spec proposal and in the prototype that we meant to contribute there would be Parent Request initiated in this function. However, there are various discussions regarding the design of this feature, so there's an ask to make this code more flexible and open for other link methods. Hence now the PR only contains the initial support for eCSL neighbor table and storing information about the wake-up peer.

Reply:
Ok, I thought that seing @abtink response but I didn't know the history. Thanks.

Reply:
I just want to restate here that in the latest draft v2 spec and discussions, the term "wake-up parent" is not even existing anymore. There's a "wake-up peer" defined instead. Better names can still be proposed in SPEC-1355, or objections to name changes. The reason for the name change is that in all typical/foreseen cases (from my viewpoint) where the WED connects to a peer device via the eCSL link, it can be handled using Link Request/Link Accept so that no Parent role is involved.   And in case the WED is directed to connect to its actual Parent, or find a Parent in the Thread Network, then it uses a regular Parent/Child resync or attach and doesn't use the special eCSL link.
The option to connect on an eCSL link using Parent Request/Response is now defined as a "future" method i.e. we have defined a bit in the wake-up frame to signal this capability, but we don't use the bit yet until we can clearly say why it's needed.

So the question still stands: do we want to comply to the spec regarding naming? If so, how long to wait at most until the naming discussion is settled? (calls are being set up but this can easily take a few more weeks). If no, why would we let this diverge?